src/vm/syncblk.inl

```diff
@@ -159,6 +160,17 @@ inline AwareLock::EnterHelperResult ObjHeader::EnterObjMonitorHelperSpin(Thread*
     }
 
     return AwareLock::EnterHelperResult_Contention;
+#else
+    // non-Windows
+    for (;;)
```

Comment:
This looks like it would effectively be an infinite spin with no yield or pause in the loop, and never having the thread go into a wait state, or did I miss something?